src/ParametricOptInterface.jl

```diff
@@ -691,6 +779,57 @@ function MOI.add_constraint(
     end
 end
 
+function MOI.delete(model::ParametricOptimizer, v::MOI.VariableIndex)
+    pop!(model.variables, v)
+    MOI.delete(model.optimizer, v)
+    return
+end
+
+function MOI.delete(
+    model::ParametricOptimizer,
+    c::MOI.ConstraintIndex{F,S},
+) where {F<:MOI.ScalarAffineFunction,S<:MOI.AbstractSet}
+    haskey(model.affine_constraint_cache, c) &&
```

Comment:
Prefer
```
if foo
    bar()
end
```
over `foo && bar()` since the latter is harder to read and code-coverage doesn't work well.